src/disco/FixedWidthField/Real/ENDF/src/write.hpp

```diff
@@ -10,22 +10,68 @@ write( Representation real, Iterator& it ){
   } else if ( isInvalid( real ) ){
     writeInvalid( it );
   } else {
-    std::array< char, w + 3 > buffer;
-    auto greedyExponent = requiresTwoDigitExponent( real );
-    const auto length = w + 2;
-    const auto decimalDigits = 6 - greedyExponent;
-    int consumedCharacters = snprintf( &buffer[0], length + 1, "%*.*E",
-                                       length, decimalDigits, real );
-    if ( consumedCharacters < 0 ){
-      throw std::exception();
+
+    // Minimal exponent size is 2 (one sign and one digit)
+    constexpr int MIN_EXPWIDTH = 2;
+    // Minimal exponent for which we can use notation
+    constexpr int MIN_EXPONENT = -2;
+
+    // Decompose real
+    double significand{ real };
+    int exponent{ 0 };
+    int expWidth{ MIN_EXPWIDTH };
+    if( real != 0.0 ){
+      exponent = std::floor( std::log10( std::abs( real ) ) );
+      if( 0 < exponent ){
+        significand /= std::pow( 10.0, exponent );
```

Comment:
I had assumed (hoped) that whatever the issue was, had been resolved. I've run this version of disco in ENDFtk without any changes and everything passed.

Reply:
I do not remember what the issue was, it could have been at read time or at write time (if it was at read time, we're actually safe since we do not use the corresponding read routine in disco). Still, I remember that that function was there for a reason so we should be vigilant, just in case.